beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestWithInArrearGreedySubscriptions.java

```diff
@@ -0,0 +1,130 @@
+        return super.getConfigSource(null, allExtraProperties);
+    }
+
+    @BeforeMethod(groups = "slow")
+    public void beforeMethod() throws Exception {
+        if (hasFailed()) {
+            return;
+        }
+        super.beforeMethod();
+        invoiceConfig.setInArrearMode(InArrearMode.GREEDY);
+
+    }
+    @Test(groups = "slow")
+    public void testWithBillRun_A() throws Exception {
+        final DateTime initialCreationDate = new DateTime(2022, 1, 13, 0, 0, 0, 0, testTimeZone);
+        clock.setTime(initialCreationDate);
+
+        final Account account = createAccountWithNonOsgiPaymentMethod(getAccountData(1));
+
+        final PlanPhaseSpecifier spec = new PlanPhaseSpecifier("basic-support-monthly-notrial", null);
+
+        busHandler.pushExpectedEvents(NextEvent.CREATE, NextEvent.BCD_CHANGE, NextEvent.BLOCK, NextEvent.NULL_INVOICE);
+        final UUID entitlementId = entitlementApi.createBaseEntitlement(account.getId(), new DefaultEntitlementSpecifier(spec, 13, null, null), "bundleExternalKey", null, null, false, true, ImmutableList.<PluginProperty>of(), callContext);
+        assertListenerStatus();
+        final Entitlement entitlement = entitlementApi.getEntitlementForId(entitlementId, callContext);
+
+        // 2022-02-01
+        clock.addDays(19);
+        busHandler.pushExpectedEvents(NextEvent.INVOICE, NextEvent.INVOICE_PAYMENT, NextEvent.PAYMENT);
+        invoiceUserApi.triggerInvoiceGeneration(account.getId(), clock.getUTCToday(), callContext);
+        assertListenerStatus();
+
+        invoiceChecker.checkInvoice(account.getId(), 1, callContext,
+                                    new ExpectedInvoiceItemCheck(new LocalDate(2022, 1, 13), new LocalDate(2022, 2, 13), InvoiceItemType.RECURRING, new BigDecimal("100.00")));
+
+
+        // 2022-02-13
+        busHandler.pushExpectedEvents(NextEvent.NULL_INVOICE);
+        clock.addDays(12);
+        assertListenerStatus();
+
+        // 2022-03-01
+        clock.addDays(16);
+        busHandler.pushExpectedEvents(NextEvent.INVOICE, NextEvent.INVOICE_PAYMENT, NextEvent.PAYMENT);
+        invoiceUserApi.triggerInvoiceGeneration(account.getId(), clock.getUTCToday(), callContext);
+        assertListenerStatus();
+
+        invoiceChecker.checkInvoice(account.getId(), 2, callContext,
+                                    new ExpectedInvoiceItemCheck(new LocalDate(2022, 2, 13), new LocalDate(2022, 3, 13), InvoiceItemType.RECURRING, new BigDecimal("100.00")));
+
+        // 2022-03-13
+        busHandler.pushExpectedEvents(NextEvent.NULL_INVOICE);
+        clock.addDays(12);
+        assertListenerStatus();
+
+        // Cancel 2022-03-13
+        busHandler.pushExpectedEvents(NextEvent.CANCEL, NextEvent.BLOCK, NextEvent.NULL_INVOICE);
+        entitlement.cancelEntitlementWithPolicyOverrideBillingPolicy(EntitlementActionPolicy.IMMEDIATE, BillingActionPolicy.IMMEDIATE, ImmutableList.of(), callContext);
+        assertListenerStatus();
+
+        // Cancel 2022-03-13
```

Comment:
Comment mistake, will fix.

Reply:
Fixed in https://github.com/killbill/killbill/pull/1824/commits/038cb86de4ae59f86d385a7b11e090e3274a09c8